samples/api/swapchain_recreation/swapchain_recreation.h

```diff
@@ -0,0 +1,195 @@
+/* Copyright (c) 2023, Google
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * Licensed under the Apache License, Version 2.0 the "License";
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#pragma once
+
+#include "common/vk_common.h"
+#include "platform/application.h"
+#include "vulkan_sample.h"
+
+/**
+ * @brief A sample that implements best practices in handling present resources and swapchain
+ * recreation, for example due to window resizing or present mode changes.
+ */
+class SwapchainRecreation : public vkb::VulkanSample
+{
+	struct SwapchainObjects
```

Comment:
To have consistent naming on those structs, maybe rename them to `SwapchainData`, `PerFrameData`, `SwapchainCleanupData`, `PresentData` ?

Reply:
At that point `Data` is just redundant; all structs are data!

If anything, it's probably more meaningful to `s/SwapchainCleanupData/SwapchainCleanupObjects`

Reply:
> If anything, it's probably more meaningful to s/SwapchainCleanupData/SwapchainCleanupObjects

Sounds reasonable as well, yes.
What made me open up this comment is `PresentOperation`, which sounds a bit strange for a struct.

Reply:
Ok I hear it (it sounds like it's for doing something, rather than holding data). The struct represents a "present operation" in the history of presents, I don't have a good name for it though really; `PresentOperationData` or `PresentOperationInfo` might do, but again `Data` and `Info` feels redundant :shrug: 

I have no strong opinion here though, I can always `sed` if you feel strongly about this.

Reply:
You're right, `Data` or `Info` would be kind of redundant. But as we are used from vulkan, structures are not named `*Create`, but `*CreateInfo`, at least for easier separation from the actual operation.
That is, ending on `Info` here would be a good fit, I think. Even though it's redundant.